membership-common/src/main/scala/com/gu/monitoring/SafeLogger.scala

```diff
@@ -1,21 +1,33 @@
 package com.gu.monitoring
 
-import com.typesafe.scalalogging.StrictLogging
-import org.slf4j.{Marker, MarkerFactory}
+import com.gu.monitoring.SafeLogger.LogMessage
+import com.typesafe.scalalogging.Logger
+import org.slf4j.{LoggerFactory, Marker, MarkerFactory}
 
-object SafeLogger extends StrictLogging {
+trait SafeLogging {
+
+  protected val logger: SafeLoggerImpl =
+    new SafeLoggerImpl(Logger(LoggerFactory.getLogger(getClass.getName)))
+
+  implicit class Sanitizer(val sc: StringContext) {
+    def scrub(args: Any*): LogMessage = {
```

Comment:
For another PR - but I wonder if we should have `scrub` as the default, and you can have `unscrubbed` logs when you explicitly set it?

Reply:
I'm not sure what you mean - scrub is a string interpolator that returns both the literal and the interpolated string, and the SafeLogger  (for `error`) sends only the literal to sentry (i.e. no PII) and the actual full log statementn to cloudwatch.
So sentry is always scrubbed, and cloudwatch is never scrubbed.

Reply:
Ah, OK - I think I misunderstood what was happening. 

Still feels like we do a lot of `logger.warn(scrub"message")` and wondered if scrubbing could just be the default.

Reply:
yeah I know what you mean, but actually by default there are no `${...}` insertions in string literals.

The usual `s"..."` is syntactic sugar for calling a (scala library) function called `s` on `StringContext` that takes the "chunks" of string literal and a list of values to insert, and returns a plain string
https://www.scala-lang.org/api/current/scala/StringContext.html

So if we want to save the normal string (for cloudwatch) as well as a scrubbed one (for sentry), we need to call a different function that returns both (e.g. in a case class). This is our function `scrub`.

We could try to somehow overload `s` or use implicits even more cleverly to have both the built in `s` and an `s` that returns the scrubbed string, but I wonder if that would be more confusing?  I think normal overloading only works on parameters.

Another option is to stop sending errors to sentry from the server side, as as far as I know no one acts on the alerts.

Happy to pair or take suggestions!